crates/oxc_traverse/src/lib.rs

```diff
@@ -144,12 +143,9 @@ mod compile_fail_tests;
 
 pub fn traverse_mut<'a, Tr: Traverse<'a>>(
     traverser: &mut Tr,
-    allocator: &'a Allocator,
     program: &mut Program<'a>,
-    symbols: SymbolTable,
-    scopes: ScopeTree,
+    mut ctx: TraverseCtx<'a>,
 ) -> (SymbolTable, ScopeTree) {
-    let mut ctx = TraverseCtx::new(scopes, symbols, allocator);
```

Comment:
I'll let @overlookmotel decide on this change.

Reply:
Please could you revert this to as it was?

`Traverse` is stacked full of unsafe code, and the scheme which maintains soundness is quite finely balanced. It'd be easy to accidentally open the door to UB. The problem is that it's a while since I wrote it, and I didn't document it enough (bad bad me!), so I'd need to retrace my steps and scratch my head a lot to figure out if it's safe to expose `TraverseCtx` or not.

As this change is not essential, it'd be preferable just to leave it as was if you don't mind please.

Reply:
In fact, this was a last resort, because prior to `traverse_mut`, the `x1_react` in `TransformerImpl` requires the `ast` from `traverse_ctx`.